src/modules/auth/passport/oidc-strategy.ts

```diff
@@ -39,15 +41,24 @@ passport.use(
       );
     }
 
+    let oidcConfig;
+
+    try {
+        oidcConfig = await new AuthService().getOIDCConfig();
+    } catch (e) {
+        logger.error(e);
+        req.res?.sendStatus(500);
```

Comment:
It seems that errors/exceptions might not have been implemented correctly from the get-go (might be my fault oops sorry).

Take a look at the [Configuration](https://www.passportjs.org/concepts/authentication/downloads/html/) section of passportjs. To do this properly, instead of manually setting the response, the `callback` function should be used to either return
- the user data `callback(null, user)`
- that the authentication failed `callback(null, false)`
- that an error occured `callback(err)`

If you have time, could you check this out? Otherwise I can take a look at it later to also check if this is done properly for the other strategies.

Reply:
Done!